### PR TITLE
podman: mirror information

### PR DESCRIPTION
**Pull request by jvanz**
Updates the podman info command to show the mirror information

Issue https://github.com/containers/libpod/issues/4553

Signed-off-by: José Guilherme Vanz <jvanz@jvanz.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Doesn't podman info man page need to be updated?

**Comment by giuseppe** [comment]
LGTM

**Comment by vrothberg** [comment]
> Doesn't podman info man page need to be updated?

Yes, the example is referencing the old format.

**Comment by jvanz** [comment]
> Doesn't podman info man page need to be updated?

Fixed

File: install.md [hunk 1]
```diff
@@ -149,6 +149,12 @@ sudo apt-get install \
   uidmap
 ```
 
+On openSUSE Leap:
+
+```bash
+sudo zypper -n in libseccomp-devel libgpgme-devel
+```
+
```
Comment (TomSweeneyRedHat) [inline install.md]: A note for the future.  This is a nice change and I appreciate you doing it, but I'd rather have it in a separate PR as it's unrelate to the podman info changes.  No worries for this PR though

File: libpod/runtime.go [hunk 2]
```diff
@@ -691,24 +691,15 @@ func (r *Runtime) Info() ([]define.InfoData, error) {
...
 	if err != nil {
 		return nil, errors.Wrapf(err, "error getting registries")
 	}
-	registries["insecure"] = ireg
-
-	breg, err := sysreg.GetBlockedRegistries()
-	if err != nil {
-		return nil, errors.Wrapf(err, "error getting registries")
+	for _, reg := range regData {
+		registries[reg.Prefix] = reg
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: WIth this in my registries.conf (comments from file removed for brevity):

```
[registries.search]
registries = ['docker.io', 'registry.fedoraproject.org', 'registry.access.redhat.com', 'registry.centos.org', 'quay.io']

[registries.insecure]
registries = ['bad.docker.io']

[registries.block]
registries = ['block.docker.io']

```

I'm not seeing the registries in the search table:

```
registries:
  bad.docker.io:
    Blocked: false
    Insecure: true
    Location: bad.docker.io
    MirrorByDigestOnly: false
    Mirrors: []
    Prefix: bad.docker.io
  block.docker.io:
    Blocked: true
    Insecure: false
    Location: block.docker.io
    MirrorByDigestOnly: false
    Mirrors: []
    Prefix: block.docker.io
```
Should it be showing all?

File: install.md (hunk 1 above)
Comment (jvanz) [reply install.md]: Fixed! :)

File: libpod/runtime.go (hunk 2 above)
Comment (jvanz) [reply libpod/runtime.go]: Fixed

File: libpod/runtime.go [hunk 3]
```diff
@@ -691,24 +691,35 @@ func (r *Runtime) Info() ([]define.InfoData, error) {
...
+	regs, err := sysreg.GetRegistries()
 	if err != nil {
 		return nil, errors.Wrapf(err, "error getting registries")
 	}
-	registries["insecure"] = ireg
-
-	breg, err := sysreg.GetBlockedRegistries()
-	if err != nil {
-		return nil, errors.Wrapf(err, "error getting registries")
+	if len(regs) > 0 {
```
Comment (vrothberg) [inline libpod/runtime.go]: `GetRegistries()` returns a slice of unqualified-search registries which is not an indicator for the config format. In fact, there is no indicator in the API at all to know if the parsed config file was in v1 or v2 format. `pkg/registries` is fully based on `containers/image/pkg/sysregistriesv2` which can parse both, v1 and v2, config files.

I think that the `else` branch is what we should be using here.

Comment (jvanz) [reply libpod/runtime.go]: Ok. So should I revert to my previous version and add the search table in addition to the registries data?

Comment (vrothberg) [reply libpod/runtime.go]: Very good point! Yes, we need to add the search registries as well. To have a consistent naming, I would name the field `unqualified-search-registries` (see https://github.com/containers/image/blob/master/pkg/sysregistriesv2/system_registries_v2.go#L139). Thanks!

Comment (jvanz) [reply libpod/runtime.go]: Something like this:
```
registries:
  bad.docker.io:
    Blocked: false
    Insecure: true
    Location: bad.docker.io
    MirrorByDigestOnly: false
    Mirrors: []
    Prefix: bad.docker.io
  block.docker.io:
    Blocked: true
    Insecure: false
    Location: block.docker.io
    MirrorByDigestOnly: false
    Mirrors: []
    Prefix: block.docker.io
  search:
  - docker.io
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - quay.io
```

Comment (jvanz) [reply libpod/runtime.go]:
```
registries:                                                                                    
  docker.io:                                         
    Blocked: true                  
    Insecure: true           
    Location: docker.io      
    MirrorByDigestOnly: false
    Mirrors:               
    - Insecure: true                              
      Location: example2.io/example/ubi8-minimal                                               
    Prefix: docker.io         
  redhat.com:                      
    Blocked: false     
    Insecure: false
    Location: registry.access.redhat.com/ubi8
    MirrorByDigestOnly: true                             
    Mirrors:      
    - Insecure: false
      Location: example.io/example/ubi8-minimal
    - Insecure: true   
      Location: example3.io/example/ubi8-minimal
    Prefix: redhat.com  
```

Comment (rhatdan) [reply libpod/runtime.go]: That looks good, now we need something similar in Buildah...

**Comment by rhatdan** [comment]
LGTM
Waiting for @vrothberg to give final LGTM

**Comment by jvanz** [comment]
I've assigned @baude because the bot said that is necessary to complete the pull request. But maybe @vrothberg is more aware of the change.
/assign @vrothberg

**Pull request merged**